File: app/lunastream/models/Product.scala
package lunastream.models

import play.api.libs.json.{Json, Format, Writes}
import java.time.LocalDateTime

case class Product(
  id: Int,
  ean: String,
  name: String,
  description: String,
  price: Option[BigDecimal],
  assembled: Boolean,
  weight: Double,
  dimension: Dimension,
  date: Option[LocalDateTime]
)

object Product {
  implicit val format: Format[Product] = Json.format[Product]
}

case class StoreProduct(
  store: Int,
  product: Product
)

object StoreProduct {
  implicit val format: Format[StoreProduct] = Json.format[StoreProduct]
}

case class PagedProducts(
  total: Long,
  from: Long,
  products: Seq[Product]
)

object PagedProducts {
  implicit val writes: Writes[PagedProducts] = Json.writes[PagedProducts]
}

case class TopProduct(
  numberOfSells: Long,
  product: Product
)


object TopProduct {
  implicit val writes: Writes[TopProduct] = Json.writes[TopProduct]
}

File: client/src/main/models/product.ts
export interface Product {
  id: number
  ean: string
  name: string
  description: string
  price?: number
  assembled: boolean
  weight: number
  dimension: Dimension
  date?: Date
}

export interface PagedProducts {
  total: number,
  from: number,
  products: Array<Product>
}

export interface TopProduct {
  numberOfSells: number
  product: Product
}

interface Dimension {
  width: number
  depth: number
  height: number
}
